File: frontend/src/api/email.ts
import request from '../utils/request'

export interface Email {
  id: string
  user_id: string
  from: string
  to: string
  subject: string
  body: string
  status: string
  is_read: boolean
  created_at: string
  updated_at: string
}

export interface SendEmailRequest {
  to: string
  subject: string
  body: string
}

export interface ListEmailsResponse {
  emails: Email[]
  total: number
  page: number
  size: number
}

export const emailApi = {
  send: (data: SendEmailRequest) =>
    request.post<any, Email>('/emails/send', data),
  
  getInbox: (page = 1, pageSize = 20, filterEmail?: string) =>
    request.get<any, ListEmailsResponse>('/emails/inbox', { 
      params: { page, page_size: pageSize, filter_email: filterEmail } 
    }),
  
  getSent: (page = 1, pageSize = 20, filterEmail?: string) =>
    request.get<any, ListEmailsResponse>('/emails/sent', { 
      params: { page, page_size: pageSize, filter_email: filterEmail } 
    }),
  
  getUnreadCount: () =>
    request.get<any, { count: number }>('/emails/unread-count'),
  
  markAsRead: (id: string) =>
    request.put(`/emails/${id}/read`),
  
  delete: (id: string) =>
    request.delete(`/emails/${id}`),
}

File: backend/internal/models/email.go
package models

import "time"

type Email struct {
	ID        string    `json:"id" gorm:"primaryKey"`
	UserID    string    `json:"user_id" gorm:"type:varchar(64);index"` // 用户ID
	From      string    `json:"from" gorm:"type:varchar(255)"`
	To        string    `json:"to" gorm:"type:varchar(255)"`
	Subject   string    `json:"subject" gorm:"type:varchar(500)"`
	Body      string    `json:"body" gorm:"type:text"`
	Status    string    `json:"status" gorm:"type:varchar(20)"` // sent, received, draft
	IsRead    bool      `json:"is_read" gorm:"default:false"`   // 已读/未读
	CreatedAt time.Time `json:"created_at"`
	UpdatedAt time.Time `json:"updated_at"`
}

type SendEmailRequest struct {
	To      string `json:"to" binding:"required,email"`
	Subject string `json:"subject" binding:"required"`
	Body    string `json:"body" binding:"required"`
}
